CandidateExercise/ConfigurationExcerciseTests/ConfigTests.cs

```diff
@@ -41,22 +70,117 @@ public void InitializeTest()
         [TestMethod]
         public void LoadValueTest()
         {
-            KVPList list = new List<KeyValuePair<string, ConfigValue>>();
-            _persistSource.Setup(ps => ps.LoadValues(It.IsAny<List<string>>())).Returns(list);
+            _log.Info("Load Value Test");
+            _persistSource.Setup(e => e.LoadValues(It.IsAny<List<string>>())).Returns(kvplist);
+
+            var convertedConfigValue = _config.GetValue("Key1");
+            Assert.AreEqual("1", convertedConfigValue.Value);
+        }
+
+        [TestMethod]
+        public void SaveValue_ShouldSaveAndVerifyTheValueToPersistentSource()
+        {
+            _log.Info("Save Value Test");
+
+            var stringConfigValue = new ConfigValue
+            {
+                Value = "Googles",
+                ValueType = ValueType.ValueString
+            };
+
+            var kvp = new KeyValuePair<string, ConfigValue>("Search", stringConfigValue);
+            _config.SaveValue(kvp);
+
+            _persistSource.Verify(e => e.PersistValues(It.Is<KVPList>(f => f[0].Key.Equals("Search"))), Times.Once);
+        }
+
+        [TestMethod]
+        public void SaveValues_ShouldSaveAndVerifyTheValuesToPersistentSource()
+        {
+
+            _log.Info("Save Values Test");
+
+            _config.SaveValues(kvplist);
+
+            _persistSource.Verify(e => e.PersistValues(It.Is<KVPList>(list => list.Intersect(kvplist).Count() == 3)), Times.Once);
+        }
+
+        [TestMethod]
+        public void GetValue_GetsAndVerifyTheValueFromPersistentSource()
+        {
+             _log.Info("Get Value Test");
+           
+
+            KVPList kvpList = new List<KVP> { new KVP(key, configValue) };
+
+            _persistSource.Setup(ps => ps.LoadValues(It.IsAny<KList>())).Returns(kvpList);
+
+            var configVal = _config.GetValue(key);
 
+            Assert.AreEqual(kvpList[0].Value, configVal);
         }
 
         [TestMethod]
-        public void SaveValueTest()
+        public void GetValues_GetsAndVerifyTheValuesFromPersistentSource()
         {
-            _log.Info("SaveValueTest");
 
-            // do the save operation
+            _log.Info("Get User Values Test");
+            _persistSource.Setup(ps => ps.LoadValues(It.IsAny<List<string>>())).Returns(kvplist);
 
-            // check for the persist call
+            var convertedConfigValues = _config.GetValues(keyList);
+            Assert.IsTrue(convertedConfigValues.Intersect(kvplist).Count() == 3);
+        }
+
+
+        [TestMethod]
+        public void SaveUserValue_ShouldSaveAndVerifyTheValueForAUserToPersistentSource()
+        {
+
+            _log.Info("Save User Value Test");
+
+            _config.SaveValues(user, kvplist);
+
+            //_persistSource.Verify(e => e.PersistValues(It.Is<KVPList>(e => e.Where(e=>e.Value.User.Equals(user))).ToList()+-+, Times.Once);
```

Comment:
Looks like you can remove this commented out code, or toss a 'TODO' in there if you have something to come back to.